src/github.com/getlantern/balancer/strategy.go

```diff
@@ -0,0 +1,106 @@
+package balancer
+
+import (
+	"math/rand"
+)
+
+// Random strategy gives even chance to each dialer, act as a baseline to other
+// strategies.
+func Random(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		// we don't need good randomness, skip seeding
+		if rand.Intn(2) == 0 {
+			return false
+		}
+		return true
+	}}
+}
+
+// Sticky strategy always pick the dialer with largest consecutive success
+// count or the smallest consecutive failure count
+func Sticky(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		mi := dialers[i].metrics()
+		mj := dialers[j].metrics()
+		return (mi.consecSuccesses - mi.consecFailures) >
```

Comment:
One special case is when local network just recovers, both have positive `consecFailures` and zero `consecSuccesses`, the one with smaller `consecFailures` is more likely reachable before network was down.